### PR TITLE
Refactors Markings, Adds Tail (some body and head) markings, Adds Alt heads, Secondary (F)Hair Themes, Darkens Vulp/Taj + Unathi horns/frills, Sprite adjustments and more...

### PR DESCRIPTION
**Pull request by KasparoVy**
**_CREDITS:_**
Tail marking idea and Vox tail marking sprites by @Fullofskittles.
Most of the new body and head marking sprites by @Fullofskittles.
5 new Tajaran hair sprites from **Polaris(?)**.
Sharp Unathi snout inspired by **TG**.

## IMPORTANT NOTICE:
**Vulpkanin and Tajara body sprites have been darkened! Characters will have to adjust their body colours if they want to look the same.**
- _Add 27 to the R, G and B values if Vulpkanin._
- _Add 15 to the R, G and B values if Tajara._

**Tiger body markings have been darkened! Characters will have to adjust their body marking colours if they want to look the same.**
- _Add 14 to the R, G and B values if body/head marking has 'tiger' in it._

**Vulpkanin facial fur patterns have been darkened! Characters will have to adjust their head accessory/facial hair colours if they want to look the same.**
- _Add 23 to the R, G and B values of your facial hair/head accessories._

**Unathi frills and horns have been darkened! Characters will have to adjust their facial hair colours if they want to look the same.**
- _Add 27 to the R, G and B values of your frills. You're going to have to do the webbing colour from scratch._
- _Add 21 to the R, G and B values of your horns._

**Tajaran head/facial hair has been darkened! Characters will have to adjust their head/facial hair colours if they want to look the same.**
- _Add 28 to the R, G and B values of your head/facial hair._

**Vulpkanin head hair has been darkened! Characters will have to adjust their head hair colours if they want to look the same.**
- _If Adhara, add 12 to R, G and B values for hair colour._
- _If Anita, add 10 to R, G and B values for hair colour._
- _If Apollo, add 6 to R, G and B values for hair colour._
- _If Belle, add 6 to R, G and B values for hair colour._
- _If Bun, add 5 to R, G and B values for hair colour._
- _If Curl, add 15 to R, G and B values for hair colour._
- _If Hawk, don't do anything._
- _If Jagged, add 10 to R, G and B values for hair colour._
- _If Kajam, add 16 to R, G and B values for hair colour._
- _If Keid, add 27 to R, G and B values for hair colour._
- _If Kleeia, add 14 to R, G and B values for hair colour._
- _If Mizar, add 14 to R, G and B values for hair colour._
- _If Short, add 7 to R, G and B values for hair colour._
- _If Spike, add 6 to R, G and B values for hair colour._

## Features, fixes and changes:
:cl:
tweak: Refactors markings. Now split into head, body and tail markings.
tweak: Refactors morph again. Changes made while morphing are now reflected instantly on your sprite.
tweak: Darkens Vulpkanin and Tajara for improved colour fidelity. Species base colour changed to compensate. Add 27 to R, G and B values if Vulp, 15 if Taj to keep the same colour on your character.
tweak: Darkens Vulpkanin facial fur patterns for improved colour fidelity. Add 23 to the R, G and B values of your facial hair/head accessory to keep the same colour on your character.
tweak: Darkens Unathi frills and horns. Add 27 to the R, G and B values of your frills and 21 for horns to keep the same colour on your character.
tweak: Vulpkanin tails are now split-rendered. Tails overlap hair when facing north, but are overlapped by hair when facing all other directions. No other differences, only affects the fluffy husky tail.
tweak: The heads of tiger markings have been isolated and can now be coloured independently.
tweak: Hair can now cover facial hair, and glasses are no longer covered by facial hair and head accessories. Means that two-piece Vulpkanin facial fur patterns won't almost entirely cover whatever glasses they're wearing.
tweak: Tajaran head and facial hair have been darkened. Add 28 to the R, G and B values of your Tajara head/facial hair to keep the same colour on your character.
tweak: Vulpkanin head hair has been darkened. Refer to PR notes for darkening factors.
fix: Naked Humans and Skrell will no longer have exposed flesh poking out their wrists when facing east or west. Male Humans now no longer lack a butt pixel when facing east or west.
fix: Mutated limbs now render correctly as soon as they're mutated.
fix: Species with TAIL_OVERLAPPED will no longer wag tail at max speed when facing north.
fix: Species with TAIL_OVERLAPPED and body accessories won't crash the game when they wag.
fix: Tiger Head and Face markings adjusted so they don't look ugly on certain species' ears.
fix: Fixes issue where Unathi dorsal stripe didn't render.
fix: Tajaran ears are no longer in their hair sprites, but on their actual heads.
fix: The adminbus body accessory now overrides the TAIL_OVERLAPPED flag and will now be rendered correctly on a body regardless of species bodyflags.
rscadd: Adds some head markings for Tajara and Vulpkanin.
rscadd: Adds body markings for Tajara, Vulpkanin and Unathi.
rscadd: Adds tail markings for Vox and Vulpkanin.
rscadd: Adds 5 Tajaran hairstyles.
rscadd: Adds a hairstyle for Humans, Unathi and Vulpkanin.
rscadd: Adds secondary hair (head and facial) themes. Skrell can now colour their tentacle cloths independently, the beads in the Tajaran/Hippie braid hairstyles can be coloured and the same with the webbing in Unathi frills. Can be changed at nanomirrors, via morph and CMA.
rscadd: Adds alternate heads. Unathi can now choose a head with a sharp snout. Can be changed via morph and CMA.
/:cl:

## Required SQL Changes:
```
ALTER TABLE `characters` CHANGE COLUMN `marking_style_name` `marking_styles` mediumtext NOT NULL DEFAULT 'head=None&body=None&tail=None';
ALTER TABLE `characters` DROP COLUMN `markings_blue`, DROP COLUMN `markings_green`;
ALTER TABLE `characters` CHANGE COLUMN `markings_red` `marking_colours` mediumtext NOT NULL DEFAULT 'head=%23000000&body=%23000000&tail=%23000000';
UPDATE `characters` SET `hair_style_name` = "Bald" WHERE `hair_style_name` = "Tajaran Ears";
ALTER TABLE `characters` ADD COLUMN `alt_head_name` varchar(45) NOT NULL AFTER head_accessory_style_name;
UPDATE `characters` SET `hair_style_name` = "Bald" WHERE `hair_style_name` = "None";
ALTER TABLE `characters` ADD COLUMN `secondary_facial_red` smallint(4) NOT NULL AFTER facial_blue;
ALTER TABLE `characters` ADD COLUMN `secondary_facial_green` smallint(4) NOT NULL AFTER secondary_facial_red;
ALTER TABLE `characters` ADD COLUMN `secondary_facial_blue` smallint(4) NOT NULL AFTER secondary_facial_green;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_red` smallint(4) NOT NULL AFTER hair_blue;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_green` smallint(4) NOT NULL AFTER secondary_hair_red;
ALTER TABLE `characters` ADD COLUMN `secondary_hair_blue` smallint(4) NOT NULL AFTER secondary_hair_green;
```

## More pictures coming.
_Vulpkanin Darkening_
vulpdarkening

_The reason why Vulpkanin tails are now split-rendered._
vulptailsplit

_Vulpkanin chargen._
markingsvulp

_Unathi chargen. The 'colour # 2' options only pop up if the selected hair/facial hair style has secondary themes and can be coloured._
unathichargen

_Unathi in-game feat. hair with secondary theme (beads) and facial hair with secondary theme (webbing)._
unathiingameid

_Vulpkanin in-game feat. hair with secondary theme (beads), chest crest and tail markings._
vulpingame

_Vulpkanin Tail-marking anims working as intended._
vulptailmarkingwag

_Vox Tail-marking anims working as intended. No more top-speed wag when facing north._
voxtailmarkingsstaticvoxtailmarkingwagvoxtailmarkingswag

_Seeing that we're able to change tail markings in the admin CMA panel._
cmatailmarkings

_Seeing that we're able to change LOTS of stuff in the admin CMA panel. Keep in mind that you can change secondary hair and secondary facial hair colours at a SalonPro Nanomirror(tm) and that you can change all this via Morph as well._
unathiadmincma

_Skrell can now independently colour their tentacle cloths. Was going to be part of another PR but the idea fit in much better here._
skrelltentaclesclothcolour

_Darkened Tajara as well in the same manner as Vulpkanin._
tajdarkening

_Tattoos using Humans as an example species._
tatsonhumans

_Hippie braid hairstyle (including the OG on Tajara. Also showcases some of the other new cosmetic features.)_
hippiebraid

_The new Tajaran hair styles from Polaris._
_Top row: L - Curly, Mid - Ladies' Retro, R - Victory Curls_
tajaranhair
_Bottom row: L - Bob, Mid - Finger Curls_

_Foxes! Showcasing more of the markings._
foxes
foxwag1foxwag2

_The rest of the Vulpkanin tail markings._
othertailmarkings
silverfoxwag1silverfoxwag3silverfoxwag2

_Glasses are no longer covered by facial hair and head accessories. You can now see glasses on Vulpkanin._
glassesfix

### CODE REVIEW

**Comment by KasparoVy** [comment]
More pictures to be added

File: code/_onclick/hud/robot.dm [hunk 1]
```diff
@@ -1,241 +1,241 @@
-/obj/screen/robot
```
Comment (KasparoVy) [inline code/_onclick/hud/robot.dm]: I have no idea why this happened. It deletes and readds everything with no differences, and I never even opened this file.

Comment (Fox-McCloud) [inline code/_onclick/hud/robot.dm]: something went wrong

**Comment by Fox-McCloud** [comment]
How is this going to impact those who already have the exact color we like for our characters? I took quite a while tweaking my character to be the proper color--will I have to re-tweak it, again, to get it back to that?

**Comment by KasparoVy** [comment]
For body colour, it's now 25 lum darker for vulpkanin-- so if you add 27 to each RGB value, it should get you back to your colour. I think it's the same number for Tajara.

**Comment by Aurorablade** [comment]
I demand taja tail jewelery

**Comment by KasparoVy** [comment]
Absolutely possible.

**Comment by FlattestGuitar** [comment]
Now you're crapping on Jay's parade. He and his snowflakey tail tattoo will no longer be so speshul.

**Comment by Fox-McCloud** [comment]
there totally needs to be a tail tip marking for the default Vulp tail

totally not biased here or anything :wink:

**Comment by KasparoVy** [comment]
@Fox-McCloud Yup, I've got it set up. Only reason why I've got that list of commands, haha. I have it in order of what I entered, as well. Also, way ahead of you. There's already a tail tip marking for the default Vulp tail in this PR, one of the things I need a picture for. No guarantees on quality, but it fits the bill.

**Comment by Aurorablade** [comment]
I SAID TAIL JEWLERY NOT TATOOS..AS LONG AS WE DO NOT PUT IN TAIL TATOOS WE ARE GOOD...

**Comment by KasparoVy** [comment]
I know what you meant and I've got some ideas, no worries.

**Comment by Jaywings** [comment]
M-more Tajaran stuff...my dream come true.

**Comment by Aurorablade** [comment]
*pulls jay into the cat box fort..*

**Comment by Lady-Luck** [comment]
Can unathi get more hair styles? Pretty much every unathi woman goes around with beehive.

**Comment by Aurorablade** [comment]
as a unathi...plz no no more unathi hair....

**Comment by Lady-Luck** [comment]
>as a unathi...plz no no more unathi hair....

Uh... why? You got all the women using the same hairstyle.

**Comment by Aurorablade** [comment]
yeah for the few of them that do you hair..I don't..we have plenty of horn and frill styles to choose from.

**Comment by Lady-Luck** [comment]
People who want hair on their unathi will continue wanting hair on their unathi, so more styles would be good so all the people who want hair on their unathi have more than one feminine hairstyle to choose from.

**Comment by Aurorablade** [comment]
....we could use more horn and frill styles though..i mean i LIKE my ramhorns..buth..

**Comment by KasparoVy** [comment]
I get what you mean, but if you're referring to porting more Unathi hair styles from a place like Polaris, Bay or Aurora, we don't have any less hair styles Unathi can use than them. I think a good idea would be to make a thread, see what hair styles people'd like most for Unathi, then work from there.

**Comment by Lady-Luck** [comment]
>I get what you mean, but if you're referring to porting more Unathi hair styles from a place like Polaris, Bay or Aurora, we don't have any less hair styles Unathi can use than them. I think a good idea would be to make a thread, see what hair styles people'd like most for Unathi, then work from there.

Hmm. I'll ask some spriters how the hair system works, maybe I could just make some minor adjustments to get some basic human hairstyles converted. If I did this, would you add them?

**Comment by KasparoVy** [comment]
Absolutely, yeah. I've already added one (the hippie braid) in a similar manner, so I've certainly no problem doing it for others

**Comment by Fox-McCloud** [comment]
@KasparoVy 

Get your butt on skype, want to run a few things by you. :stuck_out_tongue:

**Comment by Aurorablade** [comment]
No. Unathi. Hair. Please

**Comment by Lady-Luck** [comment]
No one's forcing you to use them.

**Comment by Aurorablade** [comment]
Yeah  but I would be forced to look at it like a hot pink fur color. I could accept it if the hair looked like quills or spines

**Comment by KasparoVy** [comment]
@Aurorablade Hair on Unathi = wig, spines = legit? It's certainly an idea to hold onto, I think it'd make for some interesting sprites.

**Comment by Lady-Luck** [comment]
You're already looking at hair Aurorablade, unathi can have hair. It just gives more options for the hair-users.

**Comment by Aurorablade** [comment]
Well I disagree with hair on unathi

**Comment by Spacemanspark** [comment]
I disagree with anything that is bright neon pink.
The PR itself looks nice.

**Comment by KasparoVy** [comment]
Computer crashed while I was working and totally corrupted my entire GIT history, diagram and all that. Taking a bit of time to wipe away the debris and clone my fork again.

**Comment by KasparoVy** [comment]
More pictures added. Pics of the Tajaran hairstyles are next and then I'll move on to markings

**Comment by KasparoVy** [comment]
Tajaran hair style pictures added.

**Comment by Fox-McCloud** [comment]
@KasparoVy 

Have pics of the default Vulp tail with tail tip marking?

**Comment by KasparoVy** [comment]
I'll whip one up right now, sorry for the wait.

**Comment by KasparoVy** [comment]
foxes

foxwag2foxwag1

**Comment by KasparoVy** [comment]
@Fox-McCloud
foxes

foxwag2foxwag1

**Comment by Fox-McCloud** [comment]
Thanks!

File: code/modules/mob/living/carbon/human/update_icons.dm [hunk 2]
```diff
@@ -299,7 +299,7 @@ var/global/list/damage_icon_parts = list()
 
 	//END CACHED ICON GENERATION.
 	stand_icon.Blend(base_icon,ICON_OVERLAY)
-	if(src.species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs...
+	if(((body_accessory && istype(body_accessory, /datum/body_accessory/tail)) || !body_accessory) && species.bodyflags & TAIL_OVERLAPPED) // If the user's species is flagged to have a tail that needs to be overlapped by limbs... (having a non-tail body accessory like the snake body will override this)
```
Comment (tigercat2000) [inline code/modules/mob/living/carbon/human/update_icons.dm]: um
shouldn't this just be
```
if((!body_accessory || istype(body_accessory, /datum/body_accessory/tail)) && species.bodyflags & TAIL_OVERLAPPED)
```

Comment (KasparoVy) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Is the istype okay without a check to see if body_accessory exists beforehand? Haven't had to deal with that before and it worked just fine the way it currently is. I'll test it and change if it this other way works, thanks for pointing it out

**Comment by Fox-McCloud** [comment]
Just let us know when this is ready for review.

**Comment by KasparoVy** [comment]
Ready for review. All the sprites are in order, all the lists are populated correctly, CMA and morph seem to be doing their stuff just fine and everyone who should be accessing the new features can so I guess that's all I've got unless I missed a spot. I'll keep looking, myself.

**Comment by KasparoVy** [comment]
Going to have to compensate for the parallax merge/schema fix, commit coming shortly

**Comment by KasparoVy** [comment]
Thought this was going to conflict with parallax, quite relieved it didn't. Updated and tested just to make sure, and all seems well.

**Comment by KasparoVy** [comment]
Now that's REALLY all I've got. That last bit was something that I noticed was wrong IG and back while I was doing the screenshots.

**Comment by KasparoVy** [comment]
Vulp hairstyles
---------------------------
Adhara - Darkened by 12
Anita - Darkened by 10
Apollo - Darkened by 6
Belle - Darkened by 6
Bun - Darkened by 5
Curl - Darkened by 15
Hawk - Not darkened.
Jagged - Darkened by 10
Kajam - Darkened by 16
Keid - Darkened by 27
Kleeia - Darkened by 14
Mizar - Darkened by 14
Short - Darkened by 7
Spike - Darkened by 6

**Comment by KasparoVy** [comment]
That's it, I SWEAR. Missed that while I was darkening the bodies. I'm certain now that I've forgotten nothing else. :sweat_smile: 

**Vulpkanin head hair has been darkened! Characters will have to adjust their head hair colours if they want to look the same.**
- _If Adhara, add 12 to R, G and B values for hair colour._
- _If Anita, add 10 to R, G and B values for hair colour._
- _If Apollo, add 6 to R, G and B values for hair colour._
- _If Belle, add 6 to R, G and B values for hair colour._
- _If Bun, add 5 to R, G and B values for hair colour._
- _If Cur, add 15 to R, G and B values for hair colour._
- _If Hawk, don't do anything._
- _If Jagged, add 10 to R, G and B values for hair colour._
- _If Kajam, add 16 to R, G and B values for hair colour._
- _If Keid, add 27 to R, G and B values for hair colour._
- _If Kleeia, add 14 to R, G and B values for hair colour._
- _If Mizar, add 14 to R, G and B values for hair colour._
- _If Short, add 7 to R, G and B values for hair colour._
- _If Spike, add 6 to R, G and B values for hair colour._

**Comment by Allfd** [comment]
@KasparoVy 
Really looking forward to the flexibility this will bring.

File: code/_onclick/hud/robot.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/_onclick/hud/robot.dm]: This will still need to be addressed; I recommended deleting this file, outright, then downloading it directly off Para's git website and putting it back in to see if it resolves this.

Comment (KasparoVy) [inline code/_onclick/hud/robot.dm]: Alright I'll get on that now, I did it with raw paste and that didn't work so I'll try this method too

**Comment by taukausanake** [comment]
> fix: Tajaran ears are no longer in their hair sprites, but on their actual heads.

Whoa I missed this part. I tried doing this myself one time and I didn't like how it looked. It works on Vulpkanin but for Tajaran it looks kind of bad. I don't know exactly why but I think it has to do with Tajaran being more short-hair than Vulpkanin are. I mean most of Taj hair styles are small fuzzies that don't take the full of their head while Vulp hairs are a lot longer. I'd ask to look into this more

**Comment by KasparoVy** [comment]
The hair sprites have the holes cut out for the Tajara ears so there shouldn't be any issue with hair hiding their ears if that's what your wondering. If you refer to the screenshots featuring Tajara you'll see what I mean: It looks unchanged except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour and requiring a separate head accessory to be overlain ontop of it configured with, again, the character's body colour in order to achieve the desired effect *gasp*

**tl;dr - See cat screenshots**

**Comment by KasparoVy** [comment]
The hair sprites have the holes cut out for the Tajara ears so there shouldn't be any issue with hair hiding their ears if that's what your wondering. If you refer to the screenshots featuring Tajara you'll see what I mean: It looks unchanged except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour and requiring a separate head accessory to be overlain ontop of it configured with, again, the character's body colour in order to achieve the desired effect *gasp*

**tl;dr - See cat screenshots**

@Fox-McCloud Your suggestion to replace the file worked, thanks! I navigated to https://raw.githubusercontent.com/ParadiseSS13/Paradise/master/code/_onclick/hud/robot.dm and saved the webpage as a .dm file, deleted the one in my local branch and dropped in the downloaded version and it sorted it out. Confirmed down one file change in the 'files' tab :+1:

**Comment by taukausanake** [comment]
> except the ears are now attached to the body and inherit body colour rather than being a part of a hair style and inheriting hair colour

This is what I mean. On Vulps this looks fine, different hair and ear colors. With Tajs it didn't look nice to me. Maybe it doesn't matter though but that's just what I know

**Comment by KasparoVy** [comment]
@Taukausanake Then there's still head accessories that let you colour the ears specifically in 3 different ways so you can colour the ears whatever way you like

**Comment by taukausanake** [comment]
Huh. Yeah I guess there was that. Okay, forgot or didn't see its usage back then

**Comment by KasparoVy** [comment]
No worries I'm glad you asked, it's good to clear that sort of stuff up just in case others were wondering too

**Comment by KasparoVy** [comment]
There, much better. Changed the 'required SQL changes' section as necessary

File: code/modules/mob/living/carbon/human/appearance.dm [hunk 3]
```diff
@@ -72,44 +54,59 @@
...
+	if(location == "tail")
+		stop_tail_wagging()
+	else
+		update_markings()
 	return 1
 
 /mob/living/carbon/human/proc/change_body_accessory(var/body_accessory_style)
 	var/found
-	if(!body_accessory_style)
+	if(!body_accessory_style || (src.body_accessory && src.body_accessory.name == body_accessory_style))
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/human/appearance.dm]: `src.` isn't needed, and shouldn't ever be, assuming coders don't name their proc arguments the same as their variables.
